adx/resource_adx_table_ingestion_batching_policy.go

```diff
@@ -0,0 +1,103 @@
+package adx
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"encoding/json"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type TableIngestionBatchingPolicy struct {
+	MaximumBatchingTimeSpan string
+	MaximumNumberOfItems int
+	MaximumRawDataSizeMB int
+}
+
+func resourceADXTableIngestionBatchingPolicy() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceADXTableIngestionBatchingPolicyCreate,
+		ReadContext:   resourceADXTableIngestionBatchingPolicyRead,
+		DeleteContext: resourceADXTableIngestionBatchingPolicyDelete,
+		UpdateContext: resourceADXTableIngestionBatchingPolicyCreate,
+
+		Schema: map[string]*schema.Schema{
+			"database_name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: stringIsNotEmpty,
+			},
+
+			"table_name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew: 		  true,
+				ValidateDiagFunc: stringIsNotEmpty,
+			},
+
+			"max_batching_timespan": {
+				Type:             schema.TypeString,
+				Required: true,
+				ValidateDiagFunc: stringMatch(
+					regexp.MustCompile("\\d\\d:\\d\\d:\\d\\d"),
+					"batching timespan must be in the format HH:MM:SS of ex. 00:10:00 for 10 minutes",
+					),
+			},
+
+			"max_number_items": {
```

Comment:
My assumption is that this argument represent max items per batch? Would something like `max_items_per_batch` be more self-explanatory?

Reply:
I was trying to be consistent with param names in the ADX command: https://docs.microsoft.com/en-us/azure/data-explorer/kusto/management/alter-table-ingestion-batching-policy

If we change max_items_per_batch will we also need to change max_raw_size_mb to max_raw_size_mb_per_batch?

Reply:
Then maybe just max_items?

Reply:
updated